File: MyLanguageWeb/src/main/java/ru/herooo/mylanguageweb/dto/entity/customer/response/CustomerResponseDTO.java
package ru.herooo.mylanguageweb.dto.entity.customer.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import ru.herooo.mylanguageweb.controllers.json.LongDeserializer;
import ru.herooo.mylanguageweb.controllers.json.LongSerializer;
import ru.herooo.mylanguageweb.dto.entity.country.response.CountryResponseDTO;
import ru.herooo.mylanguageweb.dto.entity.customerrole.response.CustomerRoleResponseDTO;

import java.time.LocalDateTime;

public class CustomerResponseDTO {

    @JsonSerialize(using = LongSerializer.class)
    @JsonDeserialize(using = LongDeserializer.class)
    @JsonProperty("id")
    private long id;

    @JsonProperty("nickname")
    private String nickname;

    @JsonProperty("path_to_avatar")
    private String pathToAvatar;

    @JsonProperty("description")
    private String description;

    @JsonProperty("role")
    private CustomerRoleResponseDTO role;

    @JsonProperty("date_of_create")
    private LocalDateTime dateOfCreate;

    @JsonProperty("date_of_last_visit")
    private LocalDateTime dateOfLastVisit;

    @JsonProperty("country")
    private CountryResponseDTO country;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getNickname() {
        return nickname;
    }

    public void setNickname(String nickname) {
        this.nickname = nickname;
    }

    public CustomerRoleResponseDTO getRole() {
        return role;
    }

    public void setRole(CustomerRoleResponseDTO role) {
        this.role = role;
    }

    public LocalDateTime getDateOfCreate() {
        return dateOfCreate;
    }

    public void setDateOfCreate(LocalDateTime dateOfCreate) {
        this.dateOfCreate = dateOfCreate;
    }

    public LocalDateTime getDateOfLastVisit() {
        return dateOfLastVisit;
    }

    public void setDateOfLastVisit(LocalDateTime dateOfLastVisit) {
        this.dateOfLastVisit = dateOfLastVisit;
    }

    public CountryResponseDTO getCountry() {
        return country;
    }

    public void setCountry(CountryResponseDTO country) {
        this.country = country;
    }

    public String getPathToAvatar() {
        return pathToAvatar;
    }

    public void setPathToAvatar(String pathToAvatar) {
        this.pathToAvatar = pathToAvatar;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}

File: MyLanguageWeb/src/main/resources/static/js/classes/dto/entity/customer/response/customer_response_dto.js
import {
    CustomerRoleResponseDTO
} from "../../customer_role/response/customer_role_response_dto.js";

import {
    CountryResponseDTO
} from "../../country/response/country_response_dto.js";

export class CustomerResponseDTO {
    #id;
    #nickname;
    #pathToAvatar;
    #description;
    #dateOfCreate;
    #dateOfLastVisit;
    #role;
    #country;

    constructor(json) {
        if (json) {
            this.#id = json["id"];
            this.#nickname = json["nickname"];
            this.#pathToAvatar = json["path_to_avatar"];
            this.#description = json["description"];

            let dateOfCreate = json["date_of_create"];
            if (dateOfCreate) {
                this.#dateOfCreate = new Date(dateOfCreate);
            }

            let dateOfLastVisit = json["date_of_last_visit"];
            if (dateOfLastVisit) {
                this.#dateOfLastVisit = new Date(dateOfLastVisit);
            }

            let role = json["role"];
            if (role) {
                this.#role = new CustomerRoleResponseDTO(role);
            }

            let country = json["country"];
            if (country) {
                this.#country = new CountryResponseDTO(country);
            }
        }
    }

    getId() {
        return this.#id;
    }

    getNickname() {
        return this.#nickname;
    }

    getPathToAvatar() {
        return this.#pathToAvatar;
    }

    getDescription() {
        return this.#description;
    }

    getDateOfCreate() {
        return this.#dateOfCreate;
    }

    getDateOfLastVisit() {
        return this.#dateOfLastVisit;
    }

    getRole() {
        return this.#role;
    }

    getCountry() {
        return this.#country;
    }
}
